cbor2/encoder.py

```diff
@@ -240,6 +265,12 @@ def encode_undefined(encoder, value):
     (CBORTag, encode_semantic)
 ])
 
+canonical_encoders = OrderedDict([
```

Comment:
Canonical encoding involves more than just different encoding of dicts. The RFC says that floats must be encoded to the smallest size they can fit into. Your PR does not do this.

Reply:
Good point, I had assumed we were already using the smallest representation

Reply:
Sure, but this implementation has no canonicalization whatsoever for floats. If there is another CBOR implementation that does canonicalization, it might be worth taking a look at.

Reply:
I think this implementation would be the first in the python world. I will add a minimizing float encoder and check other implementations. We should probably warn users that canonical storage of many floats will really kill performance.

For my own protocols I usually decide on a static scaling factor and just encode as an integer to avoid this whole problem.

Reply:
can this be closed now? I've changed the float encoding to be a bit more efficient...